src/SickSafetyscannersRos.cpp

```diff
@@ -83,24 +87,13 @@ SickSafetyscannersRos::SickSafetyscannersRos()
     m_laser_scan_publisher, m_diagnostic_updater, frequency_param, timestamp_param));
   m_diagnostic_updater.add("State", this, &SickSafetyscannersRos::sensorDiagnostics);
 
-  m_device = std::make_shared<sick::SickSafetyscanners>(
-    boost::bind(&SickSafetyscannersRos::receivedUDPPacket, this, _1),
-    &m_communication_settings,
-    m_interface_ip);
-  m_device->run();
-  readTypeCodeSettings();
-
-  if (m_use_pers_conf)
+  // Configure the lidar. On failure, start a timer to try again later.
+  // TODO(carlos-m159): right now configure will always return true, even
+  // if socket creation fails.
```

Comment:
There appears to be a try-catch that returns false. But I'm not sure if socket creation failure is one of the conditions that will throw an exception.